staging/src/k8s.io/legacy-cloud-providers/azure/azure.go

```diff
@@ -851,6 +858,13 @@ func (az *Cloud) updateNodeCaches(prevNode, newNode *v1.Node) {
 		if _, hasExcludeBalancerLabel := newNode.ObjectMeta.Labels[v1.LabelNodeExcludeBalancers]; hasExcludeBalancerLabel {
 			az.excludeLoadBalancerNodes.Insert(newNode.ObjectMeta.Name)
 		}
+
+		// If not in ready state and not a newly created node, add to excludeLoadBalancerNodes cache.
+		// New nodes (tainted with "node.cloudprovider.kubernetes.io/uninitialized") should not be
```

Comment:
This seems weird.  Why would we want a node that has not been initialized included a load balancer? That seems like a bug.

Reducing the number of ARM calls (don't knwo what ARM is) is a worthy goal as long as the feature for an end user still works. This seems to make that feature not work.